server/datastore/mysql/common_mysql/common.go

```diff
@@ -36,6 +38,25 @@ type DBOptions struct {
 
 func NewDB(conf *config.MysqlConfig, opts *DBOptions, otelDriverName string) (*sqlx.DB, error) {
 	driverName := "mysql"
+
+	// Parse host and port
+	host, port, err := net.SplitHostPort(conf.Address)
+	if err != nil {
+		host = conf.Address
+		port = "3306"
+	}
+
+	// Auto-detect RDS and use IAM auth if no password is provided
+	var iamTokenGen *awsIAMAuthTokenGenerator
+	useIAMAuth := false
+	if conf.Password == "" && conf.PasswordPath == "" && conf.Region != "" {
```

Comment:
Not entirely sold on requiring the region, is it required for the other AWS integration features?

Reply:
Digging into this deeper, we're doing something similar to what I suggested above where it falls back to the default region if configured (or rather, overrides the default regions in `LoadAWSConfig` if one is provided).

For MySQL there's a bit of a quandary because, without a flag specifying that you're using RDS, we don't have a good way to determine when to try IAM auth. Your original PR handled this by parsing the server address. This is very handy when it works but introduces an amount of brittleness and edge cases that we'd rather not support. So in the absence of a specific `use_iam_auth` flag (proposed and rejected due to inconsistency with the rest of the config), the presence of `region` + absence of password is the best indicator we have that IAM auth is desired.